### ISSUE TITLE
I found maybe a bug about model.predict() and model.evaluate()

### ISSUE THREAD

**Issue by wangteng200000318** [tensorflow/tensorflow#47714]
My machine is MacBook Pro 2020 13-inch, TensorFlow version is 2.4.
When I use model.predict() and model.evaluate(), no matter which function is used first, the output result of the other function will be seriously different from expected. code show as below
```
import tensorflow as tf
import matplotlib.pyplot as plt
import pylab
import numpy as np

(x_train, y_train), (x_test, y_test) = tf.keras.datasets.mnist.load_data()
x_train = x_train.reshape((60000, 28, 28, 1))
x_test = x_test.reshape((10000, 28, 28, 1))
x_train = 2 * x_train / 255.0 - 1
x_test = 2 * x_test / 255.0 - 1

plt.imshow(x_train[0, :, :, ])
print()
pylab.show()


def build_model():
    return tf.keras.Sequential([
        tf.keras.layers.Conv2D(32, (3, 3), padding='same', activation=tf.nn.relu,
                               input_shape=(28, 28, 1)),
        tf.keras.layers.Conv2D(64, (3, 3), activation=tf.nn.relu),
        tf.keras.layers.MaxPooling2D((2, 2)),
        tf.keras.layers.Conv2D(64, (3, 3), activation=tf.nn.relu),
        tf.keras.layers.MaxPooling2D((2, 2)),
        tf.keras.layers.Flatten(),
        tf.keras.layers.Dense(49, activation=tf.nn.relu),
        tf.keras.layers.Dense(10, activation=tf.nn.softmax)
    ])


def train(x, y):
    model = build_model()
    model.summary()
    model.compile(optimizer=tf.optimizers.Adam(), loss='sparse_categorical_crossentropy',
                  metrics=['accuracy'])
    model.fit(x, y, epochs=2)
    # model.evaluate(x_test, y_test)

    # predict
    print(model.predict(x_test)[0])
    print(y_test[0])
    model.evaluate(x_test,y_test)
    # print(model.predict(x_test)[0])

    return model


model = train(x_train, y_train)
```

**Comment by superp0tat0** [comment]
They are related but not the same.
model.predict(x) give the prediction of with input data x.
model.evaluate returns a tuple (loss, metric). In your definition which is (sparse_categorical_crossentropy, accuracy)

You could code yourself to compare the sparse_categorical_crossentropy loss of the test sets and the accuracy.

**Comment by wangteng200000318** [comment]
I mean that if I use model.predict(test_x), and then I use model.evaluate(test_x,test_y), the result about accuracy is very lower than my expectation. If I don't use model.predict(test_x) before using model.evaluate(test_x,test_y), I will get a perfect accuarcy.

**Comment by wangteng200000318** [comment]
> They are related but not the same.
> model.predict(x) give the prediction of with input data x.
> model.evaluate returns a tuple (loss, metric). In your definition which is (sparse_categorical_crossentropy, accuracy)
> 
> You could code yourself to compare the sparse_categorical_crossentropy loss of the test sets and the accuracy.



> They are related but not the same.
> model.predict(x) give the prediction of with input data x.
> model.evaluate returns a tuple (loss, metric). In your definition which is (sparse_categorical_crossentropy, accuracy)
> 
> You could code yourself to compare the sparse_categorical_crossentropy loss of the test sets and the accuracy.

I mean that if I use model.predict(test_x), and then I use model.evaluate(test_x,test_y), the result about accuracy is very lower than my expectation. If I don't use model.predict(test_x) before using model.evaluate(test_x,test_y), I will get a perfect accuarcy.

**Comment by Saduf2019** [comment]
@wangteng200000318 
I ran the code shared and face a different error, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/2f174d54798b4368cc6ab4fb7c2c9b63/untitled561.ipynb), please share a colab gist with the issue reported.

**Comment by wangteng200000318** [comment]
Maybe because my machine python version is 3.8.4, and the python's version in the collar is python 3.7??? Hence, I can run plt.imshow(x_train[0, :, :, ]) in my machine. 
`https://colab.research.google.com/gist/Saduf2019/2f174d54798b4368cc6ab4fb7c2c9b63/untitled561.ipynb#scrollTo=PCPHakFhwia3`
I establish two demo. In demo1, Using model.predict(x_test) before using model.evaluate(x_test,y_test). In demo2, using model.evaluate() without model.predict(). I find the accuracy is close. But in  Apple releases the ARM64 TensorFlow2.4, the result about two demos is very different.
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/57594482/111020736-f62edf00-8402-11eb-8d7c-f358684f2d3b.png">
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/57594482/111020774-37bf8a00-8403-11eb-8b4b-7c36bb70aa99.png">

**Comment by jvishnuvardhan** [comment]
@wangteng200000318 I tried on my Mac and colab with `tf-nightly` and `TF2.4`. Didn't see any issue with `Python3.7`. 

As you mentioned that you had used `Python3.8`, Can you please try with `Python3.7` and let us know whether the issue persists there. Thanks!

**Comment by wangteng200000318** [comment]
Because I use M1 chip Mac, my development environment may not allow me to use Python 3.7. If I use **Python3.7**, I will fail to configure the environment and there will be a number of errors indicating that the installation failed. My process of configuring the environment is looking at this [tutorial](https://towardsdatascience.com/tensorflow-2-4-on-apple-silicon-m1-installation-under-conda-environment-ba6de962b3b8)

**Comment by jvishnuvardhan** [comment]
@wangteng200000318 Did you install [Mac-optimized TensorFlow and TensorFlow Addons](https://github.com/apple/tensorflow_macos)? If yes, please post them in their repo  [her](https://github.com/apple/tensorflow_macos/issues) so that some experts in that repo will resolve your issue.

Support for TF on M1-chip Mac is limited currently due to reasons mentioned [here](https://github.com/tensorflow/tensorflow/issues/44751#issuecomment-765604464). thanks!
